ledger/src/ui/test/attestation/test_attestation.c

```diff
@@ -0,0 +1,359 @@
+                          int keepprivate) {
+    assert(CX_CURVE_256K1 == curve);
+    assert(pubkey == &G_att_ctx.pub_key);
+    assert(privkey == &G_att_ctx.priv_key);
+    ASSERT_STR_N_EQUALS(PRIVATE_KEY, privkey->d, KEYLEN);
+    assert(1 == keepprivate);
+    set_public_key(pubkey, PUBLIC_KEY);
+    return 0;
+}
+
+// os mocks
+unsigned int os_endorsement_get_code_hash(unsigned char *buffer) {
+    // Should not be called for the current implementation
+    ASSERT_FAIL();
+}
+
+unsigned int os_endorsement_key2_derive_sign_data(unsigned char *src,
+                                                  unsigned int srcLength,
+                                                  unsigned char *signature) {
+    assert(src == (unsigned char *)&G_att_ctx.msg);
+    assert(srcLength == G_att_ctx.msg_offset);
+    assert(signature == APDU_DATA_PTR);
+    memcpy(signature, MSG_SIGNATURE, sizeof(MSG_SIGNATURE));
+
+    return strlen(MSG_SIGNATURE);
+}
+
+void os_perso_derive_node_bip32(cx_curve_t curve,
+                                unsigned int *path,
+                                unsigned int pathLength,
+                                unsigned char *privateKey,
+                                unsigned char *chain) {
+    assert(CX_CURVE_256K1 == curve);
+    assert(path == (unsigned int *)&G_att_ctx.path);
+    assert(privateKey == (unsigned char *)&G_att_ctx.priv_key_data);
+    ASSERT_STR_N_EQUALS(path, PUBKEY_PATH, PUBKEY_PATH_LENGTH);
+    ASSERT_STR_N_EQUALS(privateKey, PRIVATE_KEY, KEYLEN);
+    assert(NULL == chain);
+}
+
+void os_memmove(void *dst, const void *src, unsigned int length) {
+    memmove(dst, src, length);
+}
+
+// signer_authorization mocks
+sigaut_signer_t *get_authorized_signer_info() {
+    memcpy(G_signer_info.hash, SIGNER_HASH, sizeof(SIGNER_HASH));
+    G_signer_info.iteration = 9;
+    return &G_signer_info;
+}
+
+// Unit tests
+void test_reset_attestation() {
+    printf("Test reset attestation...\n");
+    memcpy(G_att_ctx.msg, "a-msg", strlen("a-msg"));
+    G_att_ctx.msg_offset = strlen("a-msg");
+    memcpy(G_att_ctx.path, "a-path", strlen("a-path"));
+    set_private_key(&G_att_ctx.priv_key, (unsigned char *)PRIVATE_KEY);
+    set_public_key(&G_att_ctx.pub_key, PUBLIC_KEY);
+
+    reset_attestation(&G_att_ctx);
+    ASSERT_STRUCT_CLEARED(att_t, G_att_ctx);
+    assert(att_stage_wait_ud_value == G_att_ctx.stage);
+}
+
+void test_get_attestation_ud_value() {
+    printf("Test ATT_OP_UD_VALUE...\n");
+    reset_attestation(&G_att_ctx);
+    *(unsigned char *)N_onboarded_ui = 1;
+    memcpy(&G_att_ctx.priv_key_data, PRIVATE_KEY, sizeof(PRIVATE_KEY));
+    G_att_ctx.stage = att_stage_wait_ud_value;
+    // CLA + INS_ATTESTATION + ATT_OP_UD_VALUE + UD_VALUE
+    set_apdu("\x80\x50\x01\x46\x8d\xa8\x7f\x6a\x85\xe6\x40\x93\x27\xe1\x17\xe8"
+             "\xc7\xd2\x11\x0c\x73\x60\x22\x26\xbb\xb5\xed\xf2\x7d\x98\xc8\xa3"
+             "\x1b\xcc\xf0");
+
+    assert(3 == get_attestation(35, &G_att_ctx));
```

Comment:
Can we get rid of the magic number here and use the return value of `set_apdu` instead?